File: python/KaranlikMadde.py
class KaranlikMadde(KuantumNesnesi, IKritik):
    def __init__(self, id, stabilite, t_seviyesi):
        super().__init__(id, stabilite, t_seviyesi)

    def analiz_et(self):
        yeni_deger = self.stabilite - 15

        if yeni_deger <= 0:
            raise KuantumCokusException(f"SİSTEM ÇÖKTÜ! ID: {self.id}")
        self.stabilite = yeni_deger

    def acil_durum_sogutmasi(self):
        yeni_deger = self.stabilite + 50
        if yeni_deger > 100:
            self.stabilite = 100
            print("Sınır değer aşılıyor, nesne maxta sogutuldu.")
        else:
            self.stabilite = yeni_deger
            print(f"Nesne sogutuldu: {self.stabilite}")

File: java/KaranlikMadde.java
public class KaranlikMadde extends KuantumNesnesi implements  IKritik
{
    public KaranlikMadde(String id, int stabilite, int tseviyesi)
    {
        super(id, stabilite, tseviyesi);
    }

    @Override
    public void analizEt() throws KuantumCokusException
    {
        int yeniDeger = getStabilite() - 15;
        if (yeniDeger <= 0)
        {
            throw new KuantumCokusException(getNesneID());
        }
        setStabilite(yeniDeger);
    }
    
    @Override
    public void AcilDurumSogutmasi()
    {   
        int yeniDeger = getStabilite() + 50;
        if (!(yeniDeger > 0 && yeniDeger< 100))
        {
            System.out.println("Sınır değer aşılıyor...");
            return;
        }
        setStabilite(yeniDeger);
        System.out.println("Bu nesne soğutuldu: " +  getStabilite());
    }
}